### ISSUE TITLE
WAF: Adding to WAF condition sets doesn't reflect in planning 

### ISSUE THREAD

**Issue by hashibot** [terraform-providers/terraform-provider-aws#462]
_This issue was originally opened by @srikiraju as hashicorp/terraform#11102. It was migrated here as part of the [provider split](https://www.hashicorp.com/blog/upcoming-provider-changes-in-terraform-0-10/). The original body of the issue is below._

<hr>

### Terraform Version
Terraform v0.8.2

### Affected Resource(s)
- aws_waf_xss_match_set
- aws_waf_size_constraint_set
- aws_waf_sql_injection_match_set
- aws_waf_size_constraint_set

If this issue appears to affect multiple resources, it may be an issue with Terraform's core, so please mention this.

### Terraform Configuration Files
```hcl
resource "aws_waf_sql_injection_match_set" "sqlcond" {
  name = "condname"
  sql_injection_match_tuples {
    text_transformation = "NONE"
    field_to_match {
      type = "QUERY_STRING"
    }
  }

  sql_injection_match_tuples {
    text_transformation = "URL_DECODE"
    field_to_match {
      type = "QUERY_STRING"
    }
  }
}
```

### Expected Behavior
Changes made to match set manually should be visible in planning

### Actual Behavior
Not visible. See steps section

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform apply`
2. Manually add additional match tuples to the match set.
3. `terraform plan` - Doesn't revert or detect that changes were made

### Important Factoids
I think this happens to a lot of stuff in WAF resources

**Comment by davidcrx** [comment]
+1

**Comment by bflad** [comment]
The fix for the original issue with `aws_waf_sql_injection_match_set` resource has been submitted: #11498

The separate issue in the above comment about the `aws_waf_web_acl` resource should have been addressed roughly a year and a half ago in [version 1.30.0 of the Terraform AWS Provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1300-august-02-2018).

**Comment by bflad** [comment]
The fix for the `aws_waf_sql_injection_match_set` resource has been merged and will release with version 2.46.0 of the Terraform AWS Provider, Thursday next week.
